pulp_2to3_migration/app/plugin/rpm/xml_utils.py

```diff
@@ -39,9 +39,11 @@ def _substitute_special_chars(template):
     """
     templatetag_map = dict((sym, name) for name, sym in TemplateTagNode.mapping.items())
     symbols_pattern = '(%s)' % '|'.join(templatetag_map.keys())
-    return re.sub(symbols_pattern,
-                  lambda mobj: '{%% templatetag %s %%}' % templatetag_map[mobj.group(1)],
-                  template)
+    return re.sub(
```

Comment:
If you look at the graph, the templating operation is fairly expensive for the small amount of work it's doing (6%).  How difficult would it be to do the substitution in regex?

Reply:
That's a good question and needs some investigation. This code is from pulp2.

IIRC,  the main problem in pulp 2 was to distinguish between the template string which needs to be substituted vs the the string which looks exactly like a template but is a part of the commet/changelog/description/etc which should not be substituted.